### PR TITLE
Add Rust package with manual tests

### PR DESCRIPTION
**Pull request by adpaco-aws**
### Description of changes: 

Adds a Rust package intended to assist in manually testing the features of the extension.

### Resolved issues:

Related #17 

### Testing:

* How is this change tested? Manually.

* Is this a refactor change? No.

### Checklist
- [x] Each commit message has a non-empty body, explaining why the change was made
- [x] Methods or procedures are documented
- [x] Regression or unit tests are included, or existing tests cover the modified code
- [x] My PR is restricted to a single feature or bugfix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: resources/test-package/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+//! 9. Run the "Get coverage info" action for "test_coverage". Check that all
+//!    lines in "test_coverage" are green. In addition, check that in
+//!    "funs::find_index":
+//!     - The first and last highlighted lines are yellow.
+//!     - The second and third highlighted lines are green.
+//!     - The remaining highlighted line is red.
+//!    Comments indicating the correct colors are available in "funs::find_index".
+mod funs;
+
+#[kani::proof]
```
Comment (celinval) [inline resources/test-package/src/lib.rs:31]: I would suggest that we mimic a user crate a bit. Maybe put all the harnesses under a module that is only active if `kani` config is on. I.e.:
```rust
#[cfg(kani)]
mod verify {
  // Harnesses
}

File: resources/test-package/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! This package is intended to assist in manually testing the features of the
```
Comment (celinval) [inline resources/test-package/src/lib.rs:4]: We should really put some time into automating these scenarios.

**Review by celinval — COMMENTED** [review]
I hope we can automate these tests sooner rather than later.

I also hope that we can add a few more scenarios, like cargo workspaces, crates with dependencies, bolero... Thus, I was wondering if we should change this crate to be inside `resources/test-crates/simple-test/...`

File: resources/test-package/src/lib.rs (hunk 1 above)
Comment (adpaco-aws) [reply resources/test-package/src/lib.rs]: Done.

**Comment by adpaco-aws** [comment]
> I also hope that we can add a few more scenarios, like cargo workspaces, crates with dependencies, bolero... Thus, I was wondering if we should change this crate to be inside `resources/test-crates/simple-test/...`

Done.

**Review by celinval — APPROVED** [review]
Thanks!

**Pull request merged**